src/lib/log-forwarding.js

```diff
@@ -0,0 +1,253 @@
+      const destinationSettings = this.logForwarding.getDestinationSettings(destination)
+      const missingSecrets = []
+      destinationSettings.forEach(e => {
+        if (e.type === SECRET_FIELD_TYPE) {
+          const secretVarName = getSecretVarName(destination, e.name)
+          if (process.env[secretVarName] !== undefined) {
+            settings[e.name] = process.env[secretVarName]
+          } else {
+            missingSecrets.push(secretVarName)
+          }
+        }
+      })
+      if (missingSecrets.length > 0) {
+        throw new Error('Required secrets are missing in environment variables: ' + missingSecrets.join(', ') + '. ' +
+          'Make sure these variables are set in .env file')
+      }
+    }
+    return new LogForwardingConfig(destination, settings)
+  }
+
+  async getServerConfig () {
+    try {
+      return convertToConfigObject(await this.logForwarding.get())
+    } catch (e) {
+      throw new Error('Incorrect log forwarding configuration on server. ' + e.message)
+    }
+  }
+
+  getSupportedDestinations () {
+    return this.logForwarding.getSupportedDestinations()
+  }
+
+  getSettingsConfig (destination) {
+    return this.logForwarding.getDestinationSettings(destination)
+  }
+
+  async updateLocalConfig (lfConfig) {
+    const destination = lfConfig.getDestination()
+    const destinationSettings = this.logForwarding.getDestinationSettings(destination)
+    const projectConfig = {
+      project: this.aioConfig.project
+    }
+
+    const nonSecretSettings = {}
+    const secretSettings = {}
+
+    const settings = lfConfig.getSettings()
+    Object.keys(settings).forEach(k => {
+      const destFieldSettings = destinationSettings.find(i => i.name === k)
+      if (destFieldSettings.type === SECRET_FIELD_TYPE) {
+        secretSettings[getSecretVarName(destination, k)] = settings[k]
+      } else {
+        nonSecretSettings[k] = settings[k]
+      }
+    })
+
+    projectConfig.project.workspace.log_forwarding = {
+      [destination]: nonSecretSettings
+    }
+    const interactive = false
+    const merge = true
+    await writeAio(projectConfig, process.cwd(), { interactive, merge })
+    await writeEnv({}, process.cwd(), { interactive, merge }, secretSettings)
+  }
+
+  isLocalConfigChanged () {
+    if (fs.pathExistsSync(process.cwd() + '/tmp/' + CHECKSUM_FILE)) {
+      const oldChecksum = fs.readFileSync(process.cwd() + '/tmp/' + CHECKSUM_FILE).toString()
+      const config = this.getLocalConfigWithSecrets()
+      const newChecksum = getChecksum(config)
+      return oldChecksum !== newChecksum
+    } else {
+      return true
+    }
+  }
+
+  async updateServerConfig (lfConfig) {
+    await this.logForwarding.setDestination(lfConfig.getDestination(), lfConfig.getSettings())
+    const checksum = getChecksum(lfConfig)
+    fs.ensureDirSync(process.cwd() + '/tmp')
+    fs.writeFile(process.cwd() + '/tmp/' + CHECKSUM_FILE, checksum, { flags: 'w' })
+  }
+}
+
+class LogForwardingConfig {
```

Comment:
is this worth an object/class?

Reply:
IMO, client code (commands) look better now. But this is probably objective.